plugin/pulpcore/plugin/download/factory.py

```diff
@@ -73,22 +73,23 @@ def _make_aiohttp_session_from_remote(self):
             sslcontext = ssl.create_default_context(cafile=self._remote.ssl_ca_certificate.name)
             if self._remote.ssl_client_key.name and self._remote.ssl_client_certificate.name:
                 sslcontext.load_cert_chain(
-                    self._remote.ssl_client_key.name,
-                    self._remote.ssl_client_certificate.name
+                    self._remote.ssl_client_certificate.name,
+                    self._remote.ssl_client_key.name
                 )
         else:
             if self._remote.ssl_client_key.name and self._remote.ssl_client_certificate.name:
                 sslcontext = ssl.create_default_context()
                 sslcontext.load_cert_chain(
-                    self._remote.ssl_client_key.name,
-                    self._remote.ssl_client_certificate.name
+                    self._remote.ssl_client_certificate.name,
+                    self._remote.ssl_client_key.name
                 )
 
         if sslcontext:
             tcp_conn_opts['ssl_context'] = sslcontext
 
-        if self._remote.ssl_validation:
-            tcp_conn_opts['verify_ssl'] = self._remote.ssl_validation
+        if not self._remote.ssl_validation:
```

Comment:
@jortel ,I think we should move this condition above the preceding one, or `sslcontext` may not get into `tcp_conn_opts`.
So the suggestion is to put L90-92  before L87.